ra/ra.go

```diff
@@ -2084,35 +2084,39 @@ func (ra *RegistrationAuthorityImpl) RevokeCertByKey(ctx context.Context, req *r
 		return nil, err
 	}
 
-	// Perform an Akamai cache purge to handle occurrences of a client
-	// previously successfully revoking a certificate, but their cache purge had
-	// unexpectedly failed. Clients can re-attempt revocation and purge the
-	// Akamai cache.
-	if errors.Is(revokeErr, berrors.AlreadyRevoked) {
-		// Don't propagate purger errors to the client.
-		_ = ra.purgeOCSPCache(ctx, cert, int64(issuerID))
-	}
-
-	// Finally check the error from revocation itself. If it was an
-	// AlreadyRevoked error, try to re-revoke the cert, in case it is revoked
-	// for a reason other than keyCompromise.
-	err = revokeErr
-	if err != nil {
-		// Error out if the error was anything other than AlreadyRevoked. Otherwise
-		// try re-revocation.
-		if !errors.Is(err, berrors.AlreadyRevoked) {
-			return nil, err
-		}
-		err = ra.updateRevocationForKeyCompromise(ctx, cert.SerialNumber, int64(issuerID))
+	if revokeErr == nil {
+		// Check the error returned from revokeCertificate itself.
+		err = revokeErr
 		if err != nil {
-			return nil, err
+			// If it was an AlreadyRevoked error, try to re-revoke the cert in case
```

Comment:
You've now made it impossible to reach line 2091: if line 2087's condition evaluates to `true`, then line 2090's condition _must_ evaluate to `false`, because they're checking the same variable.

Reply:
I think the overall structure here can just be
```go
err = revokeErr
if err == nil {
  // do happy-path purge and return
} else if errors.Is(err, berrors.AlreadyRevoked) {
  // update revocation for key compromise, purge, and return
} else {
  // error return
}
```